run/test/setupReusable.ts

```diff
@@ -180,14 +180,13 @@ describe('setupReusable', function() {
       consumeSink(sink: Producer<string>) {
         return pipe(
           sink,
-          subscribe(
-            () => {},
-            () => {
-              sinkCompleted++;
-            }
-          )
+          subscribe(() => {})
         );
       }
+
+      cleanup() {
+        sinkCompleted++;
```

Comment:
I'm trying to understand what you meant with the _shoehorn_ so I'm looking at examples. In this example, this `cleanup` function increments this variable, but I'd argue that `cleanup` has nothing to do with the `sink`, nothing to do with sink completion, it's just a function that driver authors can use for various other clean ups.

Reply:
`cleanup` is only called when `engine.dispose()` is called. `onEnd` is also called when the stream ends

Reply:
So what's the shoehorn? :thinking:

Reply:
You need a second function just because `onEnd` happens to only be called on a subset of possible termination causes.

Reply:
But different things are being terminated, and clean up logic would be unrelated to the sink.

Reply:
so you'd like `cleanup` as dual to the constructor? So the source has its own cleanup (waiting for `t==2` from downstream), the sink has its own cleanup (either `uponEnd` or `subscribe.onEnd` depending on https://github.com/cyclejs/callbags/pull/26) and the whole driver gets a third cleanup for everything unrelated to either source or sink?

Reply:
Yes!